File: src/cpp/cdm/system/equipment/mechanical_ventilator/actions/SEMechanicalVentilatorMode.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/
#include "cdm/CommonDefs.h"
#include "cdm/system/equipment/mechanical_ventilator/actions/SEMechanicalVentilatorMode.h"

SEMechanicalVentilatorMode::SEMechanicalVentilatorMode(Logger* logger) : SEMechanicalVentilatorAction(logger)
{
  m_Connection = eSwitch::Off;
}

SEMechanicalVentilatorMode::~SEMechanicalVentilatorMode()
{
  m_Connection = eSwitch::Off;
}

void SEMechanicalVentilatorMode::Clear()
{
  SEMechanicalVentilatorAction::Clear();
  m_Connection = eSwitch::Off;
}

bool SEMechanicalVentilatorMode::IsValid() const
{
  if (m_Connection == eSwitch::NullSwitch)
    return false;
  return SEMechanicalVentilatorAction::IsValid();
}

bool SEMechanicalVentilatorMode::IsActive() const
{
  return m_Connection == eSwitch::On;
}

bool SEMechanicalVentilatorMode::ToSettings(SEMechanicalVentilatorSettings& s, const SESubstanceManager& /*subMgr*/)
{
  if (!IsValid())
    return false;

  s.Clear();
  s.SetConnection(m_Connection);
  return true;
}

void SEMechanicalVentilatorMode::SetConnection(eSwitch c)
{
  m_Connection = c;
}
eSwitch SEMechanicalVentilatorMode::GetConnection() const
{
  return m_Connection;
}

File: src/java/com/kitware/pulse/cdm/system/equipment/mechanical_ventilator/actions/SEMechanicalVentilatorMode.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package com.kitware.pulse.cdm.system.equipment.mechanical_ventilator.actions;

import com.kitware.pulse.cdm.bind.Enums.eSwitch;
import com.kitware.pulse.cdm.bind.MechanicalVentilatorActions.MechanicalVentilatorModeData;
import com.kitware.pulse.cdm.properties.SEScalarPressure;
import com.kitware.pulse.cdm.properties.SEScalarVolumePerTime;

public class SEMechanicalVentilatorMode extends SEMechanicalVentilatorAction
{
  protected eSwitch connection;

  public SEMechanicalVentilatorMode()
  {
    clear();
  }

  public SEMechanicalVentilatorMode(SEMechanicalVentilatorMode other)
  {
    copy(other);
  }

  public void copy(SEMechanicalVentilatorMode other)
  {
    super.copy(other);
    connection = other.connection;
  }

  public void clear()
  {
    connection = eSwitch.Off;
  }
  
  protected static void load(MechanicalVentilatorModeData src, SEMechanicalVentilatorMode dst)
  {
    dst.reset();
    SEMechanicalVentilatorAction.load(src.getMechanicalVentilatorAction(),dst);
    dst.setConnection(src.getConnection());
  }
  protected static void unload(SEMechanicalVentilatorMode src, MechanicalVentilatorModeData.Builder dst)
  {
    SEMechanicalVentilatorAction.unload(src, dst.getMechanicalVentilatorActionBuilder());
    dst.setConnection(src.getConnection());
  }

  public boolean isValid()
  {
    return true;
  }

  public eSwitch getConnection()
  {
    return connection;
  }

  public void setConnection(eSwitch s)
  {
    connection = s;
  }

  public String toString()
  {
    String str = "Mechanical Ventilator Mode";
    str += "\n\tConnection: " + this.connection;

    return str;
  }
}
